props/saber_sabersense_buttons.h

```diff
@@ -932,28 +932,37 @@ bool Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
 #endif
 
     // BLASTER DEFLECTION
-    // 1 Button
-    // or 2 button with extra define.
-#if NUM_BUTTONS == 1 || defined(SABERSENSE_BLAST_MAIN_AND_AUX)
+    // 1 Button (and 2 button with extra define).
     case EVENTID(BUTTON_POWER, EVENT_FIRST_SAVED_CLICK_SHORT, MODE_ON):
-      // For harmonized 'Exit Colour Menu' control in OS-7.x. Ignored in OS-8.
       if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) {
+        // For harmonized 'Exit Colour Menu' control in OS-7.x. Ignored in OS-8 due to Modes.
         ToggleColorChangeMode();
-      } else {
-        // Fires blast in all OSs.
-        swing_blast_ = false;
-        SaberBase::DoBlast();
+        return true;
       }
+      
+#if NUM_BUTTONS == 1
+      // If NOT in color change mode and 1 button system, do blast.
+      swing_blast_ = false;
+      SaberBase::DoBlast();
+      return true;
+      
+#elif NUM_BUTTONS == 2
+      // If NOT in color change mode and 2 button system, do Blast only if defined.
+#ifdef SABERSENSE_BLAST_MAIN_AND_AUX
+      swing_blast_ = false;
```

Comment:
This code looks the same as the NUM_BUTTONS == 1 case.
Before you had

```
#if NUM_BUTTONS == 1 || defined(SABERSENSE_BLAST_MAIN_AND_AUX)
```

I think maybe we can use that again to unify these two pieces of code and make things easier to read at the same time?